otter/convergence/service.py

```diff
@@ -581,6 +607,23 @@ def converge(tenant_id, group_id, dirty_flag):
     yield do_return(parallel(effs))
 
 
+@do
+def get_recently_converged_groups(recently_converged, interval):
+    """
+    Return a list of recently converged groups, and garbage-collect any groups
+    in the recently_converged map that are no longer 'recent'.
+    """
+    # STM would be cool but this is synchronous so whatever
+    recent = orig = yield recently_converged.read()
+    now = yield Effect(Func(time.time))
+    for group in recent.keys():
+        if now - recent[group] > interval:
+            recent = recent.remove(group)
```

Comment:
Python `dict.keys()` returns a copy of the keys, and it'd be strange for PMap to break that contract.  I think it does return a copy:  it makes a `PVector` of all the keys (https://github.com/tobgu/pyrsistent/blob/master/pyrsistent/_pmap.py#L117).

But also, `recent` is not being mutated, the variable is just being bound to [a new PMap](https://github.com/tobgu/pyrsistent/blob/master/pyrsistent/_pmap.py#L161), which I think is fine?

```
>>> a = [2, 3, 1]
>>> for i in a:
...    print i
...    a = []
... 
2
3
1
```